swan-lake/development-tutorials/test-document-the-code/test-ballerina-code/mocking.md

```diff
@@ -110,9 +109,9 @@ import ballerina/io;
 
 http:Client clientEndpoint = check new ("https://api.chucknorris.io/jokes/");
 
-type Joke readonly & record {|
+type Joke readonly & record {
```

Comment:
why can't we have a closed record?

Reply:
Trying to map the response to a closed record not containing all fields of the response would cause a `PayloadBindingClientError`. The recommended best practice in this case is to use an open record containing the necessary fields.